### PR TITLE
refactor tocvdebug.d

### PR DESCRIPTION
**Pull request by WalterBright**
Some minor refactoring, mostly adding `const`.

### CODE REVIEW

File: src/dmd/tocvdebug.d [hunk 1]
```diff
@@ -91,31 +91,24 @@ uint cv4_memfunctypidx(FuncDeclaration fd)
...
+    if (AggregateDeclaration ad = fd.isMember2())
     {
         // It's a member function, which gets a special type record
 
-        idx_t thisidx;
-        if (fd.isStatic())
-            thisidx = dttab4[TYvoid];
-        else
-        {
-            assert(ad.handleType());
```
Comment (wilzbach) [inline src/dmd/tocvdebug.d]: Isn't this changing the behavior as now `null` could be passed to `Type_toCtype`?

https://github.com/dlang/dmd/blob/806f105da1d802919f5741ecc1568c9623e5377f/src/dmd/toctype.d#L253-L261

AFAICT there's nothing ensuring that the type of the `AggregateDeclaration` isn't null?

**Review by wilzbach — COMMENTED** [review]
LGTM except for the removed `assert(ad.handleType());`

File: src/dmd/tocvdebug.d (hunk 1 above)
Comment (WalterBright) [reply src/dmd/tocvdebug.d]: It's a reasonable point. I added it back.

**Review by RazvanN7 — APPROVED** [review]

**Review by jacob-carlborg — APPROVED** [review]

**Comment by WalterBright** [comment]
I turned off automerge because a later version of this PR was failing oddly, so I'm adding in bit by bit till I find the problem. This is why I prefer lots of smaller PRs :-)

**Review by wilzbach — APPROVED** [review]

**Comment by wilzbach** [comment]
FYI: the Jenkins failure is unrelated (ci.dlang.io is offline) and I have disabled the web hook, so the failure won't be displayed for new PRs.
The closing/re-opening seems to have worked to start SemaphoreCI.

Also auto-tester seem quite greenish now (see below) - so is this good to go?

image

**Comment by WalterBright** [comment]
Not quite ready yet.

**Comment by JinShil** [comment]
> Not quite ready yet.

Just waiting for you to declare this ready.

**Comment by WalterBright** [comment]
Sorry, I forgot. It's ready!

**Review by JinShil — APPROVED** [review]

**Pull request merged**